### PR TITLE
Fixes #130. Initialize allocatable variables in Process Library

### LINKED ISSUE #130: Changing klid causes a halt in GOCART
As discovered by @wmputman, if you change the `klid` (aka `pressure_lid_in_hPa`) in GOCART, the model locks up.

It works if `pressure_lid_in_hPa: 0.01` but `pressure_lid_in_hPa: 0.40` causes it to not work. 

I have a hotfix for this incoming.

### PR DESCRIPTION
**Pull request by mathomp4**
Closes #130 

TL;DR: This PR fixes the "changing klid halts GOCART" issue seen by @wmputman. 

My testing shows that it is zero-diff if klid is set to 0.01 (aka k=1), and lets the model run at other klid values.

Runs with:

- [x] Intel Release
- [x] Intel Debug
- [x] GNU Release
- [x] GNU Debug

---

The solution was found by running with Intel in Debug mode where it pointed to line 1412:

https://github.com/GEOS-ESM/GOCART/blob/4d4f27c51f49ccebb2df5e167582d033eea4d0d4/Process_Library/GOCART2G_Process.F90#L1400-L1414

The issue is that `vsettle` is filled by `Chem_CalcVsettle` but only for `vsettle(:,:,klid:km)`. But then this line:

```fortran
          vsettle = max(1.0e-9, vsettle - v_upwardMaring)
```

operates on *all* of `vsettle`. Since `vsettle` was only allocated and never initialized to anything, boom.

To be safe, I'm initializing *all* local allocatable arrays that I can see to zero. (I'm not a scientist, so someone like @pcolarco or @amdasilva might want to make sure that's a "safe" or "valid" initialization value to use.)

Also, in one subroutine, I changed the name of the array `dp` to `d_p`. The issue is that there  was an array:

```fortran
real(kind=DP), dimension(:,:), allocatable :: dp
```

So `dp` was `kind=DP`. The compiler sort of went a bit nuts when I tried `source=0.0_DP` in that routine. So rather than change the kind, we just change `dp` to `d_p` as was done in another routine.

### CODE REVIEW

**Review by amdasilva — DISMISSED** [review]
Looks good.

**Review by tclune — APPROVED** [review]
cmake changes ok

**Comment by mathomp4** [comment]
Releasing from draft. It seems to work with both Intel and Debug

File: Process_Library/GOCART2G_Process.F90 [hunk 1]
```diff
@@ -1183,10 +1183,10 @@ subroutine Chem_SettlingSimple ( km, klid, flag, cdt, grav, &
 
    hsurf => hghte(i1:i2,j1:j2,km)
 
-   allocate(dz(i2,j2,km), radius(i2,j2,km), rhop(i2,j2,km), vsettle(i2,j2,km), qa(i2,j2,km))
+   allocate(dz(i2,j2,km), radius(i2,j2,km), rhop(i2,j2,km), vsettle(i2,j2,km), qa(i2,j2,km), source=0.0)
    allocate(cmass_before(i2,j2), cmass_after(i2,j2))
-   cmass_before = 0.d0
-   cmass_after = 0.d0
+   cmass_before = 0.0_DP
+   cmass_after = 0.0_DP
```
Comment (adarmenov) [inline Process_Library/GOCART2G_Process.F90:1189]: [L1187-L1189](https://github.com/GEOS-ESM/GOCART/pull/131/files#diff-c654445ea9c7fe0dfe3d5bcf55254b11562955161b9315737ac00bfddba60ed4R1187-R1189) can be replaced with the equivalent: 
>  allocate(cmass_before(i2,j2), cmass_after(i2,j2), source=0.0_DP)

File: Process_Library/GOCART2G_Process.F90 [hunk 2]
```diff
@@ -1359,10 +1359,10 @@ subroutine Chem_Settling ( km, klid, bin, flag, cdt, grav, &
...
```
Comment (adarmenov) [inline Process_Library/GOCART2G_Process.F90:1365]: Similarly L1363-L1365 can be replaced with:
> allocate(cmass_before(i2,j2), cmass_after(i2,j2),source=0.0_DP)

File: Process_Library/GOCART2G_Process.F90 [hunk 3]
```diff
@@ -2459,9 +2461,9 @@ subroutine Chem_SettlingSimpleOrig ( km, klid, flag, grav, cdt, radiusInp, rhopI
      qa(i1:i2,j1:j2,klid) = qa(i1:i2,j1:j2,klid) - qdel
 
      do k = klid+1, km
-      dp   = delp(i1:i2,j1:j2,k)
+      d_p   = delp(i1:i2,j1:j2,k)
       dpm1 = delp(i1:i2,j1:j2,k-1)
```
Comment (adarmenov) [inline Process_Library/GOCART2G_Process.F90:2465]: Rename variable **dpm1** to **d_pm1** for consistency with **d_p**.

**Review by adarmenov — DISMISSED** [review]
Approved. Feedback provided.

**Comment by mathomp4** [comment]
@adarmenov Okay. Let me change that code. Give me a second...

File: Process_Library/GOCART2G_Process.F90 (hunk 1 above)
Comment (mathomp4) [reply Process_Library/GOCART2G_Process.F90]: Done. Please take a look

File: Process_Library/GOCART2G_Process.F90 (hunk 2 above)
Comment (mathomp4) [reply Process_Library/GOCART2G_Process.F90]: Done. Please take a look

**Review by adarmenov — APPROVED** [review]

**Review by tclune — DISMISSED** [review]

**Review by adarmenov — DISMISSED** [review]
